setup_hooks.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+rustup component add rustfmt-preview
+
+rustfmt_path=`find $HOME/.rustup/toolchains/ -wholename '*bin/rustfmt*'`
```

Comment:
I still do not follow why `which rustfmt` doesn't work.

Reply:
There could be multiple `rustfmt` in system. I have 5, for example. Which one should we pick?

Reply:
Isn't `which rustfmt` returns the executable one that's the default (and useful) one?

```
benzh:cv-rs benzh$ which rustfmt
/Users/benzh/.cargo/bin/rustfmt
benzh:cv-rs benzh$ find $HOME/.rustup/toolchains/ -wholename '*rustfmt*'
/Users/benzh/.rustup/toolchains//nightly-x86_64-apple-darwin/bin/rustfmt
/Users/benzh/.rustup/toolchains//nightly-x86_64-apple-darwin/lib/rustlib/manifest-rustfmt-preview-x86_64-apple-darwin
/Users/benzh/.rustup/toolchains//nightly-x86_64-apple-darwin/share/doc/rustfmt
/Users/benzh/.rustup/toolchains//nightly-x86_64-apple-darwin/share/doc/rustfmt/LICENSE-APACHE
/Users/benzh/.rustup/toolchains//nightly-x86_64-apple-darwin/share/doc/rustfmt/README.md
/Users/benzh/.rustup/toolchains//nightly-x86_64-apple-darwin/share/doc/rustfmt/LICENSE-MIT
```

Reply:
So `bin/rustfmt` also returns two, stable and nightly, if I have two installed at the same time

Reply:
AFAIK you can get it only if you spefically install two `rustfmt` with different default toolchain. Probably I am wrong? 

Anyhow, what's your proposal?

Reply:
`which rustfmt`